### PR TITLE
GNOME: 46 → 47

### PR DESCRIPTION
**Pull request by bobby285271**
This pull request updates GNOME to [version 47](https://release.gnome.org/47/), released on 2024-09-18.

This branch is last squashed and rebased on 2024-10-03, you can check the previous branch at  d3023767128d59fccf411015b66fd544bcbfba05.

Notable changes:

- FileChooser implementation is now done by nautilus.
- GJS has been rebased on mozjs 128.
- gspell has been ported to meson.
- Tracker and tracker-miners are renamed to tinysparql and localsearch.
- Tinysparql dropped libsoup 2 support.

### CODE REVIEW

**Comment by bobby285271** [comment]
Oh sorry, another force push soon...

I think I replaced all gitlab.gnome.org with gitlab.org by mistake :joy:

**Comment by bobby285271** [comment]
\* [GNOME/gobject-introspection!477](https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/477) will require https://github.com/WebKit/WebKit/pull/30446/commits/fff1b1773bff2ef7c3b867ab019d69faa36c010d.

**Comment by fabianhjr** [comment]
https://github.com/NixOS/nixpkgs/pull/343127

**Comment by fabianhjr** [comment]
There is an issue I haven't figured out but kgx / GNOME Console terminal emulator is broken. (Using other terminal like alacritty has been a workaround)

Don't really know how to diagnose the issue

image

**Comment by bobby285271** [comment]
It might take me 8+ (??) hours to get back to a NixOS setup but yes I can reproduce this on my last testing on vte 0.77.91 and gnome-console 47.beta, using zsh. If you can help with bisect or you know what to fixup that will be appreciated (to avoid rebuilds I guess you can pick this 2 commits to unstable)

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -1527,6 +1527,8 @@ mapAliases ({
   transmission-qt = lib.warn (transmission3Warning {suffix = "-qt";}) transmission_3-qt; # Added 2024-06-10
   treefmt = treefmt2; # 2024-06-28
   libtransmission = lib.warn (transmission3Warning {prefix = "lib";}) libtransmission_3; # Added 2024-06-10
+  tracker = tinysparql; # Added 2024-09-05
+  tracker-miners = localsearch; # Added 2024-09-05
```
Comment (Pandapip1) [inline pkgs/top-level/aliases.nix:1531]: Shouldn't this use `lib.warn`?

```suggestion
  tracker = lib.warn "tracker has been renamed to tinysparql" tinysparql; # Added 2024-09-05
  tracker-miners = lib.warn "tracker-miners has been renamed to localsearch" localsearch; # Added 2024-09-05
```

**Comment by fabianhjr** [comment]
Already on the commits @nyabinary: https://github.com/NixOS/nixpkgs/pull/333911/commits/9f0630293eb1aa4b023746ed092a5812354f4fd3

File: pkgs/by-name/lo/localsearch/package.nix [hunk 2]
```diff
@@ -121,11 +121,6 @@ stdenv.mkDerivation (finalAttrs: {
   mesonFlags = [
     # TODO: tests do not like our sandbox
     "-Dfunctional_tests=false"
-
-    # libgrss is unmaintained and has no new releases since 2015, and an open
-    # security issue since then. Despite a patch now being availab, we're opting
```
Comment (nyabinary) [inline pkgs/by-name/lo/localsearch/package.nix:122]: Typo here (should be available instead of availab)

**Comment by vcunat** [comment]
I assume that you strongly prefer to start with x86_64-linux, so I did that:
https://hydra.nixos.org/jobset/nixpkgs/gnome

Though not many binaries are likely to arrive today.

**Comment by bobby285271** [comment]
(Note that 2ed7a5191179fb137855551ae5949618b060765c is a mass rebuild)

**Comment by vcunat** [comment]
Rebuilding "anew" now (cancelled the obsolete builds).

File: pkgs/desktops/gnome/extensions/update-extensions.py [hunk 3]
```diff
@@ -22,6 +22,7 @@
     "44": "44",
```
Comment (SuperSandro2000) [inline pkgs/desktops/gnome/extensions/update-extensions.py:22]: Shouldn't we drop some old versions at some point?

File: pkgs/development/libraries/glibmm/2.68.nix
Comment (SuperSandro2000) [inline pkgs/development/libraries/glibmm/2.68.nix:1]: That file name is completely not matching the version it contains

File: pkgs/top-level/aliases.nix [hunk 4]
```diff
@@ -1575,6 +1575,8 @@ mapAliases {
   transmission-qt = lib.warn (transmission3Warning {suffix = "-qt";}) transmission_3-qt; # Added 2024-06-10
   treefmt = treefmt2; # 2024-06-28
   libtransmission = lib.warn (transmission3Warning {prefix = "lib";}) libtransmission_3; # Added 2024-06-10
+  tracker = lib.warn "tracker has been renamed to tinysparql" tinysparql; # Added 2024-09-30
+  tracker-miners = lib.warn "tracker-miners has been renamed to localsearch" localsearch; # Added 2024-09-30
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:1579]:
```suggestion
  tracker = tinysparql; # Added 2024-09-30
  tracker-miners = localsearch; # Added 2024-09-30
```

File: nixos/modules/services/desktops/gnome/tinysparql.nix [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  config = lib.mkIf cfg.enable {
+
+    environment.systemPackages = [ pkgs.tinysparql ];
+
+    services.dbus.packages = [ pkgs.tinysparql ];
+
+    systemd.packages = [ pkgs.tinysparql ];
+
+  };
+
```
Comment (SuperSandro2000) [inline nixos/modules/services/desktops/gnome/tinysparql.nix:79]:
```suggestion
  options = {
    services.gnome.tinysparql = {
      enable = lib.mkOption {
        type = lib.types.bool;
        default = false;
        description = ''
          Whether to enable TinySPARQL services, a search engine,
          search tool and metadata storage system.
        '';
      };
    };
  };

  config = lib.mkIf cfg.enable {
    environment.systemPackages = [ pkgs.tinysparql ];

    services.dbus.packages = [ pkgs.tinysparql ];

    systemd.packages = [ pkgs.tinysparql ];
  };
```
Especially those ### comments are legacy and can be removed/not added new with good conscience

File: nixos/modules/services/desktops/gnome/tinysparql.nix [hunk 6]
```diff
@@ -0,0 +1,80 @@
+# TinySPARQL daemon.
+
```
Comment (SuperSandro2000) [inline nixos/modules/services/desktops/gnome/tinysparql.nix:2]:
```suggestion
```

File: nixos/modules/services/desktops/gnome/localsearch.nix [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+          search engine and metadata storage system.
+        '';
+      };
+
+    };
+
+  };
+
+  ###### implementation
+
```
Comment (SuperSandro2000) [inline nixos/modules/services/desktops/gnome/localsearch.nix:53]:
```suggestion
```

File: nixos/modules/services/desktops/gnome/localsearch.nix [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+        "services"
+        "gnome"
+        "localsearch"
+        "enable"
+      ]
+    )
+  ];
+
+  ###### interface
+
```
Comment (SuperSandro2000) [inline nixos/modules/services/desktops/gnome/localsearch.nix:34]:
```suggestion
```

File: pkgs/development/libraries/glibmm/2.68.nix
Comment (bobby285271) [reply pkgs/development/libraries/glibmm/2.68.nix:1]: See https://github.com/NixOS/nixpkgs/pull/247766#discussion_r1333821411

File: pkgs/top-level/aliases.nix (hunk 4 above)
Comment (bobby285271) [reply pkgs/top-level/aliases.nix:1579]: You probably want to continue the discussion in https://github.com/NixOS/nixpkgs/pull/333911#discussion_r1777463411 :joy:

File: pkgs/desktops/gnome/extensions/update-extensions.py (hunk 3 above)
Comment (bobby285271) [reply pkgs/desktops/gnome/extensions/update-extensions.py:22]: PRs welcomed (I am not familiar with `pkgs/desktops/gnome/extensions`).

**Comment by bobby285271** [comment]
@ofborg build gupnp_1_6

**Review by fabianhjr — APPROVED** [review]
Been using gnome 47 for a while now

**Review by stigtsp — APPROVED** [review]
Perl diff LGTM

**Comment by fabianhjr** [comment]
Is it ok if I merge into staging-next?

Any extra changes could still go there before it is merged into the main branch (still about 40k darwin builds away)

**Comment by vcunat** [comment]
OfBorg says 33k darwin rebuilds, so I'd say we just integrate it into the next week's `staging-next` cycle, one way or another.

**Comment by peigongdsd** [comment]
Looks like some images are missing for the settings app. Maybe this is a bug or maybe just on my side.
image

**Comment by Pandapip1** [comment]
I'd like to test this, but I don't have a good machine to do a mass rebuild on. Is there a binary cache available?

**Comment by kirillrdy** [comment]
@peigongdsd same here 
image

**Comment by jtojnar** [comment]
> Looks like some images are missing for the settings app. Maybe this is a bug or maybe just on my side.

That looks like a regression from https://github.com/NixOS/nixpkgs/commit/5fa6b40311ca7bf45d9290c39001fdc8d59f8f31

Possibly, they changed file format of animations.

**Comment by bobby285271** [comment]
c211b5a37d3cd4b03017f6756d60754597a54b4b
https://gitlab.gnome.org/GNOME/gnome-control-center/-/merge_requests/2697

I am looking at [random](https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/5994) gst MRs and think animated WebP belongs to -bad.

**Comment by peigongdsd** [comment]
> Looks like some images are missing for the settings app. Maybe this is a bug or maybe just on my side. image

Confirm fixed
image

**Comment by jtojnar** [comment]
It is weird that it requires `gst-plugins-bad` – it was previously also using animated WebP images. Even weirder is that there are no relevant warnings in the console. Comparing the outputs of running `gnome-control-center` with `GST_DEBUG="*:3"`, the only significant thing I see is that with `bad`, a warning changes:

```patch
-FIXME  [00m [00m        videodecoder gstvideodecoder.c:1193:gst_video_decoder_drain_out:<vp9dec7>[00m Sub-class should implement drain()
+FIXME  [00m [00m        videodecoder gstvideodecoder.c:1193:gst_video_decoder_drain_out:<maindec>[00m Sub-class should implement drain()
+FIXME  [00m [00m        videodecoder gstvideodecoder.c:1193:gst_video_decoder_drain_out:<alphadec>[00m Sub-class should implement drain()
```

So possibly, it is using https://gstreamer.freedesktop.org/documentation/codecalpha/vp9alphadecodebin.html?gi-language=c from `bad`.

**Comment by cguentherTUChemnitz** [comment]
How is the branching mechanism here? Does the merge into staging mean, that this is becoming available in unstable branch of nixos packages? Or is there an additional merge between staging and unstable necessary?

**Comment by ninelore** [comment]
@cguentherTUChemnitz check these out :)

https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#staging

https://nixpk.gs/pr-tracker.html?pr=333911

**Comment by vcunat** [comment]
You can expect it there in a week or two, as a part of https://github.com/NixOS/nixpkgs/pull/348827

Staging is described in https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#staging

**Comment by peigongdsd** [comment]
Hello I'm wondering which branch should I track if I'm gonna adopt gnome 47 at this time if the gnome branch is deleted? Is binary cache available for the stating branch?

**Comment by kirillrdy** [comment]
> Hello I'm wondering which branch should I track if I'm gonna adopt gnome 47 at this time if the gnome branch is deleted? Is binary cache available for the stating branch?

gnome 47 is on current iteration of staging-next branch,  there is some binary cache but not all

**Pull request merged**